aioshelly/rpc_device/wsrpc.py

```diff
@@ -390,16 +396,17 @@ async def _rpc_call(
         future: asyncio.Future[dict[str, Any]] = self._loop.create_future()
         call = RPCCall(self._next_id, method, params, self._session, future)
         self._calls[call.call_id] = call
+        timeout_handle = self._loop.call_at(
+            self._loop.time() + timeout, _handle_timeout, future
+        )
 
         try:
-            async with asyncio.timeout(timeout):
-                await self._send_json(call.request_frame)
-                resp = await future
+            await self._send_json(call.request_frame)
```

Comment:
The problem with this approach is that the json send could block forever if we hit the high water mark